### PR TITLE
[Fix](nereids) fix normalize repeat alias rewrite

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
Induced by #34196.
In NormalizeRepeat, when NormalizeToSlot is called, aggregate function parameters, grouping scalar function parameters, and all expressions in grouping sets (including columns and column aliases) are pushed down to the lower-level project output.
In the previous PR #34196, the context was split into two, but the two contexts were not consistent. It is possible that the triplets in one context save (id, c1, id as c1), and the triplets in the other context save (id, id, id).
This causes id as c1 to be pushed down, but there is a reference to id in the upper-level LogicalRepeat, which causes the slot to be not found.
This pr has been modified.
If the same slot in the projection column has different aliases, for example, 
```sql
select id as c1, id, id as c3, grouping(id) from table1 group by grouping sets((id, value2),(id));
```
then id as c1 (using the first alias) will be pushed down to the project.
In both the LogicalRepeat operator and the LogicalAggregate operator, c1 is referenced as the input slot, and id and c3 will not be used as input slots.
```sql
before NormalizeRepeat:
LogicalResultSink[32] ( outputExprs=[c1#3, id#0, c3#4, __grouping_3#5] )
+--LogicalRepeat ( groupingSets=[[id#0, value2#2], [id#0]], outputExpressions=[id#0 AS `c1`#3, id#0, id#0 AS `c3`#4, Grouping(id#0) AS `Grouping(id)`#5] )
   +--LogicalOlapScan (qualified=table1)
After NormalizeRepeat:
LogicalResultSink[33] (outputExprs=[c1#3, id#0, c3#4, __grouping_3#5])
+--LogicalAggregate[30] (groupByExpr=[c1#3, value2#2, GROUPING_ID#7, GROUPING_PREFIX_c1#6 originExpression=Grouping(c1#3)], outputExpr=[c1#3, c1#3 AS `id`#0, c1#3 AS `c3`#4, GROUPING_PREFIX_c1#6 originExpression=Grouping(c1#3) AS `GROUPING_PREFIX_c1`#5], hasRepeat=true )
   +--LogicalRepeat (groupingSets=[[c1#3, value2#2], [c1#3]], outputExpressions=[c1#3, value2#2, GROUPING_ID#7, GROUPING_PREFIX_c1#6 originExpression=Grouping(c1#3)] )
      +--LogicalProject[28] (projects=[id#0 AS `c1`#3, value2#2])
         +--LogicalOlapScan (qualified=table1)
```

### CODE REVIEW

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by morrySnow** [comment]
add desc

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run cloud_p0

**Review by morrySnow — DISMISSED** [review]

File: regression-test/suites/nereids_rules_p0/grouping_sets/grouping_alias_test.groovy [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    sql """
+    INSERT INTO grouping_alias_test_t (id, value1, value2) VALUES
+    (1, 10, 'A'),
+    (2, 20, 'B'),
+    (3, 30, 'C'),
+    (4, 40, 'D');
+    """
+
+    qt_alias "select id as c1 from grouping_alias_test_t group by grouping sets((id,value2),(id)) order by 1;"
+    qt_alias_grouping_scalar "select id as c1, grouping(id) from grouping_alias_test_t  group by grouping sets((id,value2),(id)) order by 1,2;"
```
Comment (starocean999) [inline regression-test/suites/nereids_rules_p0/grouping_sets/grouping_alias_test.groovy:38]: we can add more test case like bellow:
` select id as c1, id, id as c3, grouping(id) from table1 group by grouping sets((c1, value2),(c3));`
the alias in grouping sets

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run p0

**Comment by feiniaofeiafei** [comment]
run external

**Review by starocean999 — APPROVED** [review]

**Comment by feiniaofeiafei** [comment]
run p0

**Comment by feiniaofeiafei** [comment]
run p0

**Review by morrySnow — APPROVED** [review]

**Pull request merged**